php/php.lua

```diff
@@ -0,0 +1,192 @@
+    -- Update URL
+    updateUrl = "https://raw.githubusercontent.com/version-fox/version-fox-plugins/main/php/php.lua",
+    -- minimum compatible vfox version
+    minRuntimeVersion = "0.2.2",
+}
+
+--- Returns some pre-installed information, such as version number, download address, local files, etc.
+--- If checksum is provided, vfox will automatically check it for you.
+--- @param ctx table
+--- @field ctx.version string User-input version
+--- @return table Version information
+function PLUGIN:PreInstall(ctx)
+    local version = ctx.version
+    if OS_TYPE == "windows" then
+        return getReleaseForWindows(version)
+    else
+        return getReleaseForLinux(version)
+    end
+end
+
+function getReleaseForWindows(version)
+    local baseUrl = "https://windows.php.net/downloads/releases/archives/"
+    local resp, err = http.get({
+        url = baseUrl
+    })
+    local doc = html.parse(resp.body)
+    local prefix = "php-" .. version .. "-nts"
+    local x86, x64 = "", ""
+    doc:find("a"):each(function(i, selection)
+        local filename = selection:text()
+        if string.sub(filename, 1, #prefix) == prefix then
+            if string.find(filename, "x86") then
+                x86 = filename
+            end
+            if string.find(filename, "x64") then
+                x64 = filename
+            end
+        end
+    end)
+
+    local url = ""
+
+    if (ARCH_TYPE == "amd64" or ARCH_TYPE == "arm64") and x64 ~= "" then
+        url = baseUrl .. x64
+    end
+    if ARCH_TYPE == "386" and x86 ~= "" then
+        url = baseUrl .. x86
+    end
+
+    if url ~= "" then
+        return {
+            version = version,
+            url = url
+        }
+    end
+end
+
+function getReleaseForLinux(version)
+    local resp, err = http.get({
+        url = URL .. "/releases/index.php?json&version=" .. version
+    })
+    local data = json.decode(resp.body)
+
+    local filename, md5, sha256 = "", "", ""
+    for _, s in ipairs(data["source"]) do
+        if string.match(s.filename, "%.tar%.gz$") then
+            filename = s.filename
+            md5 = s.md5
+            sha256 = s.sha256
+            break
+        end
+    end
+    return {
+        version = version,
+        url = URL .. "/distributions/" .. filename,
+        sha256 = sha256,
+        md5 = md5
+    }
+end
+
+--- Extension point, called after PreInstall, can perform additional operations,
+--- such as file operations for the SDK installation directory or compile source code
+--- Currently can be left unimplemented!
+function PLUGIN:PostInstall(ctx)
+    --- ctx.rootPath SDK installation directory
+    local rootPath = ctx.rootPath
+    local sdkInfo = ctx.sdkInfo["php"]
+    local path = sdkInfo.path
+    local version = sdkInfo.version
+    local name = sdkInfo.name
+    if OS_TYPE == "windows" then
+        return
+    end
+
+    local install_path = rootPath .. "/tmp-" .. version
+    os.execute("mv " .. path .. " " .. install_path)
```

Comment:
There should handle the failure of the command to execute, otherwise `vfox` will treat the installation as successful, and it will actually fail.